### ISSUE TITLE
Warnings: [general] 'NSKeyedUnarchiveFromData' should not be used to for un-archiving and will be removed in a future release

### ISSUE THREAD

**Issue by vkgupta857** [firebase/firebase-ios-sdk#11425]
### Description

Getting warning 
`[general] 'NSKeyedUnarchiveFromData' should not be used to for un-archiving and will be removed in a future release`
when Firebase messaging is added through Pod

on debugging, it was found that the line 

`tokenInfo = [NSKeyedUnarchiver unarchiveObjectWithData:item];`

of function

```
+ (nullable FIRMessagingTokenInfo *)tokenInfoFromKeychainItem:(NSData *)item {
  // Check if it is saved as an archived FIRMessagingTokenInfo, otherwise return nil.
  FIRMessagingTokenInfo *tokenInfo = nil;
  // NOTE: Passing in nil to unarchiveObjectWithData will result in an iOS error logged
  // in the console on iOS 10 and below. Avoid by checking item.data's existence.
  if (item) {
    // TODO(chliangGoogle: Use the new API and secureCoding protocol.
    @try {
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
      [NSKeyedUnarchiver setClass:[FIRMessagingTokenInfo class]
                     forClassName:@"FIRInstanceIDTokenInfo"];
      tokenInfo = [NSKeyedUnarchiver unarchiveObjectWithData:item];
#pragma clang diagnostic pop

    } @catch (NSException *exception) {
      FIRMessagingLoggerDebug(kFIRMessagingMessageCodeTokenStoreExceptionUnarchivingTokenInfo,
                              @"Unable to parse token info from Keychain item; item was in an "
                              @"invalid format");
      tokenInfo = nil;
    } @finally {
    }
  }
  return tokenInfo;
}
```
is the reason for the issue.

### Firebase SDK Version

10.3.0

### Xcode Version

14.2

### Installation Method

CocoaPods

### Firebase Product(s)

Messaging, Remote Config

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Multiple warning with same message

[general] 'NSKeyedUnarchiveFromData' should not be used to for un-archiving and will be removed in a future release
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

PODS:
  - Adjust (4.33.4):
    - Adjust/Core (= 4.33.4)
  - Adjust/Core (4.33.4)
  - Alamofire (5.4.4)
  - AlamofireNetworkActivityLogger (3.4.0):
    - Alamofire (~> 5.4.0)
  - AWSCore (2.28.5)
  - AWSRekognition (2.28.5):
    - AWSCore (= 2.28.5)
  - CryptoSwift (1.6.0)
  - FBAEMKit (16.0.1):
    - FBSDKCoreKit_Basics (= 16.0.1)
  - FBSDKCoreKit (16.0.1):
    - FBAEMKit (= 16.0.1)
    - FBSDKCoreKit_Basics (= 16.0.1)
  - FBSDKCoreKit_Basics (16.0.1)
  - FBSDKLoginKit (16.0.1):
    - FBSDKCoreKit (= 16.0.1)
  - FBSDKShareKit (16.0.1):
    - FBSDKCoreKit (= 16.0.1)
  - Firebase/CoreOnly (10.3.0):
    - FirebaseCore (= 10.3.0)
  - Firebase/Messaging (10.3.0):
    - Firebase/CoreOnly
    - FirebaseMessaging (~> 10.3.0)
  - Firebase/RemoteConfig (10.3.0):
    - Firebase/CoreOnly
    - FirebaseRemoteConfig (~> 10.3.0)
  - FirebaseABTesting (10.3.0):
    - FirebaseCore (~> 10.0)
  - FirebaseCore (10.3.0):
    - FirebaseCoreInternal (~> 10.0)
    - GoogleUtilities/Environment (~> 7.8)
    - GoogleUtilities/Logger (~> 7.8)
  - FirebaseCoreInternal (10.3.0):
    - "GoogleUtilities/NSData+zlib (~> 7.8)"
  - FirebaseInstallations (10.3.0):
    - FirebaseCore (~> 10.0)
    - GoogleUtilities/Environment (~> 7.8)
    - GoogleUtilities/UserDefaults (~> 7.8)
    - PromisesObjC (~> 2.1)
  - FirebaseMessaging (10.3.0):
    - FirebaseCore (~> 10.0)
    - FirebaseInstallations (~> 10.0)
    - GoogleDataTransport (~> 9.2)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.8)
    - GoogleUtilities/Environment (~> 7.8)
    - GoogleUtilities/Reachability (~> 7.8)
    - GoogleUtilities/UserDefaults (~> 7.8)
    - nanopb (< 2.30910.0, >= 2.30908.0)
  - FirebaseRemoteConfig (10.3.0):
    - FirebaseABTesting (~> 10.0)
    - FirebaseCore (~> 10.0)
    - FirebaseInstallations (~> 10.0)
    - GoogleUtilities/Environment (~> 7.8)
    - "GoogleUtilities/NSData+zlib (~> 7.8)"
  - FlexibleSteppedProgressBar (0.7.2)
  - GoogleDataTransport (9.2.0):
    - GoogleUtilities/Environment (~> 7.7)
    - nanopb (< 2.30910.0, >= 2.30908.0)
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleMaps (6.2.1):
    - GoogleMaps/Maps (= 6.2.1)
  - GoogleMaps/Base (6.2.1)
  - GoogleMaps/Maps (6.2.1):
    - GoogleMaps/Base
  - GoogleUtilities/AppDelegateSwizzler (7.10.0):
    - GoogleUtilities/Environment
    - GoogleUtilities/Logger
    - GoogleUtilities/Network
  - GoogleUtilities/Environment (7.10.0):
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleUtilities/Logger (7.10.0):
    - GoogleUtilities/Environment
  - GoogleUtilities/Network (7.10.0):
    - GoogleUtilities/Logger
    - "GoogleUtilities/NSData+zlib"
    - GoogleUtilities/Reachability
  - "GoogleUtilities/NSData+zlib (7.10.0)"
  - GoogleUtilities/Reachability (7.10.0):
    - GoogleUtilities/Logger
  - GoogleUtilities/UserDefaults (7.10.0):
    - GoogleUtilities/Logger
  - nanopb (2.30909.0):
    - nanopb/decode (= 2.30909.0)
    - nanopb/encode (= 2.30909.0)
  - nanopb/decode (2.30909.0)
  - nanopb/encode (2.30909.0)
  - NVActivityIndicatorView (5.1.1):
    - NVActivityIndicatorView/Base (= 5.1.1)
  - NVActivityIndicatorView/Base (5.1.1)
  - PromisesObjC (2.1.1)
  - SDWebImage (5.14.2):
    - SDWebImage/Core (= 5.14.2)
  - SDWebImage/Core (5.14.2)
  - Smartech-iOS-SDK (3.2.12)
  - SwiftAlgorithms (1.0.0)
  - SwiftLint (0.50.3)
  - TOCropViewController (2.6.1)

DEPENDENCIES:
  - Adjust (~> 4.33.3)
  - Alamofire (~> 5.4.0)
  - AlamofireNetworkActivityLogger (~> 3.4)
  - AWSRekognition (~> 2.27)
  - CryptoSwift
  - FBSDKCoreKit (~> 16.0)
  - FBSDKLoginKit (~> 16.0)
  - FBSDKShareKit (~> 16.0)
  - Firebase/Messaging
  - Firebase/RemoteConfig
  - FlexibleSteppedProgressBar
  - GoogleMaps
  - NVActivityIndicatorView
  - SDWebImage
  - Smartech-iOS-SDK (~> 3.2.11)
  - SwiftAlgorithms (~> 1.0)
  - SwiftLint
  - TOCropViewController

SPEC REPOS:
  trunk:
    - Adjust
    - Alamofire
    - AlamofireNetworkActivityLogger
    - AWSCore
    - AWSRekognition
    - CryptoSwift
    - FBAEMKit
    - FBSDKCoreKit
    - FBSDKCoreKit_Basics
    - FBSDKLoginKit
    - FBSDKShareKit
    - Firebase
    - FirebaseABTesting
    - FirebaseCore
    - FirebaseCoreInternal
    - FirebaseInstallations
    - FirebaseMessaging
    - FirebaseRemoteConfig
    - FlexibleSteppedProgressBar
    - GoogleDataTransport
    - GoogleMaps
    - GoogleUtilities
    - nanopb
    - NVActivityIndicatorView
    - PromisesObjC
    - SDWebImage
    - Smartech-iOS-SDK
    - SwiftAlgorithms
    - SwiftLint
    - TOCropViewController

SPEC CHECKSUMS:
  Adjust: 4690e07ea401ee56d4d3f4f5bdad86c94f51903f
  Alamofire: f3b09a368f1582ab751b3fff5460276e0d2cf5c9
  AlamofireNetworkActivityLogger: 162ab8aee00e6267a4304d7cc134e13ccfe3bcc5
  AWSCore: 4bad3f0eb17393b4926a9491e32b835991c56382
  AWSRekognition: aa0b988fc9cf5c88625b73ee35191cf092259fd9
  CryptoSwift: 562f8eceb40e80796fffc668b0cad9313284cfa6
  FBAEMKit: daac7466b918752f020345be5c7d9787f98cfc07
  FBSDKCoreKit: 2cb033464b2134af0138f87d20b859eb3f9be359
  FBSDKCoreKit_Basics: d37280da2e65872f0e931d15f45d97ea324fec37
  FBSDKLoginKit: c47a3b90920702487b10b647c63520dc2676dad9
  FBSDKShareKit: dbc6adab8c837275fa6e7f5f0992dbde4fea87eb
  Firebase: f92fc551ead69c94168d36c2b26188263860acd9
  FirebaseABTesting: e6660693429b4663573c82f8d2f1041deff1753a
  FirebaseCore: 988754646ab3bd4bdcb740f1bfe26b9f6c0d5f2a
  FirebaseCoreInternal: 29b76f784d607df8b2a1259d73c3f04f1210137b
  FirebaseInstallations: e2f26126089dcf41e215f7b8925af8d953c7d602
  FirebaseMessaging: e345b219fd15d325f0cf2fef28cb8ce00d851b3f
  FirebaseRemoteConfig: c24f767c17b0440ee63c7e93380d599173556113
  FlexibleSteppedProgressBar: 6a781cfed749bf59c69813a8706aa397c342ab5c
  GoogleDataTransport: 1c8145da7117bd68bbbed00cf304edb6a24de00f
  GoogleMaps: 20d7b12be49a14287f797e88e0e31bc4156aaeb4
  GoogleUtilities: bad72cb363809015b1f7f19beb1f1cd23c589f95
  nanopb: b552cce312b6c8484180ef47159bc0f65a1f0431
  NVActivityIndicatorView: 1f6c5687f1171810aa27a3296814dc2d7dec3667
  PromisesObjC: ab77feca74fa2823e7af4249b8326368e61014cb
  SDWebImage: b9a731e1d6307f44ca703b3976d18c24ca561e84
  Smartech-iOS-SDK: 1663dedfed7d786e2bcd0c34f3041664ed4fb1c2
  SwiftAlgorithms: 38dda4731d19027fdeee1125f973111bf3386b53
  SwiftLint: 77f7cb2b9bb81ab4a12fcc86448ba3f11afa50c6
  TOCropViewController: edfd4f25713d56905ad1e0b9f5be3fbe0f59c863

PODFILE CHECKSUM: d989261e7f0548b6ff47a0cd4e661f9506227d82

COCOAPODS: 1.12.1

```

</details>

**Comment by paulb777** [comment]
Firebase 10.x still supports iOS 11 which requires the usage of the deprecated method. Xcode 15 drops iOS 11 support and Firebase's next major release will drop iOS 11 support as well. We'll update this API usage at that point.

In the meantime, initial testing shows the deprecated API is still functional with Xcode 15 and iOS 17

**Comment by paulb777** [comment]
This should be fixed by #12110 that went out in today's 10.19.0 release.
